### PR TITLE
Creation wizard: fix reset

### PR DESCRIPTION
**Pull request by m4dz**
When adding a new account, the Account Creation Wizard component state isn't properly resetted. This PR forces to rely of `CLOSE_MODAL` event for resetting and improve the way the modal can be closed.

Now, a modal is systematically closed with the `CLOSE_MODAL` event, regardless where the event comes (cancel click, success click, setTimeout…). This event is used in the RouterStore to update the URL properly.

### CODE REVIEW

File: client/app/actions/account_action_creator.coffee [hunk 1]
```diff
@@ -36,9 +36,17 @@ module.exports = AccountActionCreator =
...
+                    AppDispatcher.dispatch
+                        type: ActionTypes.CLOSE_MODAL,
+                        value: {id: account.id, mailboxID: account.inboxMailbox}
+                , 5000
+
+                # We pass the timer so we can store and clear it in RouterStore
                 AppDispatcher.dispatch
                     type: ActionTypes.ADD_ACCOUNT_SUCCESS
-                    value: {account, areMailboxesConfigured}
+                    value: {account, areMailboxesConfigured, modalTimer}
```
Comment (misstick) [inline client/app/actions/account_action_creator.coffee]: Be careful of anti-pattern; Stores have nothings to do with actionCreator except listening to them. Deciding what action to do is the stuff of ... ActionCreators files :) (I know it seems obvious each time I explain it).

You should add a property to the store, such as `isWaitingForRedirections`, or maybe use `RequestsStore` that is used for this. 
Preventing action `close_modal` is not usefull; if you get the right data to know if modal is still opened or not when you listen to this event, you would not need adding `modalTimer` as an argument.

**Comment by frankrousseau** [comment]
@m4dz After the recent merges, this PR is red and cannot be merged. @m4dz could you fix it?

**Comment by m4dz** [comment]
This is on my list for today

**Comment by frankrousseau** [comment]
Great thank you @m4dz

File: client/app/router.coffee [hunk 2]
```diff
@@ -53,7 +53,7 @@ class Router extends Backbone.Router
 
 
     defaultView: ->
-        url = if (mboxID = RouterStore.getDefaultAccount().get 'inboxMailbox')
+        url = if (mboxID = RouterStore.getDefaultAccount()?.get 'inboxMailbox')
```
Comment (misstick) [inline client/app/router.coffee]: @m4dz : could you write `mailboxID` instead plz?

File: client/app/actions/account_action_creator.coffee [hunk 3]
```diff
@@ -36,10 +45,12 @@ module.exports = AccountActionCreator =
                                          account.draftMailbox? and \
                                          account.trashMailbox?
 
+                # We pass the timer so we can store and clear it in RouterStore
```
Comment (misstick) [inline client/app/actions/account_action_creator.coffee]: no more timer anymore: I think you can remove this comment ;)

File: client/app/actions/account_action_creator.coffee [hunk 4]
```diff
@@ -9,7 +9,16 @@ XHRUtils      = require '../libs/xhr'
 AccountStore = require '../stores/account_store'
 RouterStore  = require '../stores/router_store'
 
-
+###
```
Comment (misstick) [inline client/app/actions/account_action_creator.coffee]: 👍

**Comment by misstick** [comment]
@m4dz : except typo, it's OK for me: can you clean this so that I can merge it plz?

File: client/app/actions/account_action_creator.coffee (hunk 4 above)
Comment (m4dz) [inline client/app/actions/account_action_creator.coffee]: \o/

**Pull request merged**